python/strings_udf/strings_udf/_lib/cpp/strings_udf.pxd

```diff
@@ -11,9 +11,19 @@ from cudf._lib.cpp.types cimport size_type
 from rmm._lib.device_buffer cimport DeviceBuffer, device_buffer
 
 
+cdef extern from "cudf/strings/udf/udf_string.hpp" namespace \
+        "cudf::strings::udf" nogil:
+    cdef cppclass udf_string
```

Comment:
Luckily we shouldn't ever need to actually expose any methods here. The only reason even this is exposed is so that the declaration below can be written with the correct function signature, and then over in `cudf_jit_udf.pyx` so it can be correctly `sizeof`'ed, beyond that, numba doesn't ever use it. The sizes of `string_view` and `udf_string` are associated with the instances through the `StructModel`s defined in `_typing.py`, - that's where numba gets all it needs to know about the size so no need to expose those members here. As for the methods, they're only ever called thread by thread within the shim functions.